src/machine/system_calls.rs

```diff
@@ -4215,23 +4215,7 @@ impl Machine {
 
     #[inline(always)]
     pub(crate) fn maybe(&mut self) {
-        fn generate_random_bits(rng: &mut rand::rngs::StdRng, num_bits: usize) -> u64 {
-            let mut random_bits: u64 = 0;
-
-            for _ in 0..num_bits {
-                random_bits <<= 1;
-
-                if rng.gen_bool(0.5) {
-                    random_bits |= 1;
-                }
-            }
-
-            random_bits
-        }
-
-        let result = { generate_random_bits(&mut self.rng, 1) == 0 };
-
-        self.machine_st.fail = result;
+        self.machine_st.fail = (&mut self.rng).gen_range(0..=1) == 0;
```

Comment:
Why not:
```suggestion
        self.machine_st.fail = self.rng.gen_bool(0.5);
```

Reply:
`gen_bool` generates a random number from a Bernoulli sampler which does a bunch of math using floats, where `gen_range` generates it from a uniform sampler which is served directly by the rng implementation (which are all uints underneath). So I expect it to be faster.

Good call on `self.rng.gen_*`, I'll update the PR to use that.

Reply:
I got curious so I made a test on godbolt: https://godbolt.org/z/3xoneGYrz

Based on that, gen_range does seem to be shorter but my concern for floats seems to be misplaced; it looks like float instructions are optimized out.

Also I found `.gen::<bool>()` which seems to be better than both so I'll switch to that. 😅